FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -0,0 +1,1142 @@
+
+/**
+ * @brief Ticks to wait for task notifications.
+ */
+#define httpexampleDEMO_TICKS_TO_WAIT                        pdMS_TO_TICKS( 1000 )
+
+/**
+ * @brief Notification bit indicating an error between tasks.
+ */
+#define httpexampleHTTP_FAILURE                              ( 1U << 1 )
+
+/**
+ * @brief Notification bit indicating completion of the request task.
+ */
+#define httpexampleREQUEST_TASK_COMPLETION                   ( 1U << 2 )
+
+/**
+ * @brief Notification bit indicating completion of the response task.
+ */
+#define httpexampleRESPONSE_TASK_COMPLETION                  ( 1U << 3 )
+
+/**
+ * @brief The maximum number of loop iterations to wait for task completion, or
+ * after the last received server response, before declaring failure.
+ */
+#define httpexampleMAX_WAIT_ITERATIONS                       ( 5 )
+
+/**
+ * @brief The maximum number of times to run the loop in this demo.
+ *
+ * @note The loop is executed only if the demo fails initially. Once the demo
+ * loop succeeds on a single iteration, the demo exits successfully.
+ */
+#ifndef HTTP_MAX_DEMO_LOOP_COUNT
+    #define HTTP_MAX_DEMO_LOOP_COUNT    ( 3 )
+#endif
+
+/**
+ * @brief Time in ticks to wait between retries of the demo loop, if
+ * demo loop fails.
+ */
+#define DELAY_BETWEEN_DEMO_RETRY_ITERATIONS_TICKS    ( pdMS_TO_TICKS( 5000U ) )
+
+/* Each compilation unit must define the NetworkContext struct. */
+struct NetworkContext
+{
+    TlsTransportParams_t * pParams;
+};
+
+/**
+ * @brief The network context used for the TLS session with the server.
+ */
+static NetworkContext_t xNetworkContext;
+
+/**
+ * @brief The host address string extracted from the pre-signed URL.
+ *
+ * @note httpexampleS3_PRESIGNED_GET_URL_LENGTH is set as the array length here
+ * as the length of the host name string cannot exceed this value.
+ */
+static char cServerHost[ httpexampleS3_PRESIGNED_GET_URL_LENGTH ];
+
+/**
+ * @brief The length of the host address found in the pre-signed URL.
+ */
+static size_t xServerHostLength;
+
+/**
+ * @brief The location of the path within the pre-signed URL.
+ */
+static const char * pcPath;
+
+/**
+ * @brief Data type for the request queue.
+ *
+ * Contains the request header struct and its corresponding buffer, to be
+ * populated and enqueued by the request task, and read by the main task. The
+ * buffer is included to avoid pointer inaccuracy during queue copy operations.
+ */
+typedef struct RequestItem
+{
+    HTTPRequestHeaders_t xRequestHeaders;
+    uint8_t ucHeaderBuffer[ democonfigUSER_BUFFER_LENGTH ];
+} RequestItem_t;
+
+/**
+ * @brief Data type for the response queue.
+ *
+ * Contains the response data type and its corresponding buffer, to be enqueued
+ * by the main task, and interpreted by the response task. The buffer is
+ * included to avoid pointer inaccuracy during queue copy operations.
+ */
+typedef struct ResponseItem
+{
+    HTTPResponse_t xResponse;
+    uint8_t ucResponseBuffer[ democonfigUSER_BUFFER_LENGTH ];
```

Comment:
The default memory requirement for the demo seems really high. 20(default Queue size)*~4K(default buffer size) * 2(number of queues). As discussed offline, please try to optimize this requirement if possible